src/Storage/Device/S3.php

```diff
@@ -686,6 +686,12 @@ private function call(string $method, string $uri, string $data = '', array $par
         }
 
         $this->headers['date'] = \gmdate('D, d M Y H:i:s T');
+        
+        // if cn zone selected, switch to amazonaws.cn
+        if ($this->region == 'cn-north-4' || $this->region == 'cn-northwest-1') {
+            $this->headers['host'] = $this->bucket . '.s3.'.$this->region.'.amazonaws.cn';
+        }
```

Comment:
Please make this change in the constructor rather than this method to ensure it won't interfere with any other subclass.